### PR TITLE
Rendering tweaks

### LINKED ISSUE #58: lines have spikes (specially in higher map zool level) when there are many segments
#### When drawing path with many segments, some spikes appears. Please take a look at the image below:
image

After digging a bit, it seems the issue indeed is because of the lineJoin settings of the Canvas. The default is "miter".

## Proposed solution:
If I add the following line to drawPath function inside canvas-display.js, it would solve this issue
```
this.ctx.lineJoin = 'round';
```
##### Here is what it looks after adding the line above:
image

I am suggesting to add the line below to drawPath in canvas-display.js 
```
this.ctx.lineJoin = attrs['lineJoin'] || 'miter';
```
image

and the line below in the render function inside renderedsegment.js
```
display.drawPath(this.renderData, {
        fill: 'none',
        stroke: styler.compute2('segments', 'stroke', this),
        'stroke-width': styler.compute2('segments', 'stroke-width', this),
        'stroke-dasharray': styler.compute2('segments', 'stroke-dasharray', this),
        'stroke-linecap': styler.compute2('segments', 'stroke-linecap', this),
        'lineJoin':'round',
      });
```
image

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR makes the following changes:
- Allows specifying font families and font sizes for segment labels and place labels in the styles property,
- Render the background of segment labels with a size consistent with the font size specified in the styles property (a method that consistently formats the font size string is added behind the scenes),
- Fix #58 (jagged edges with spikes),
- Updated Storybook stories.

### CODE REVIEW

File: lib/util/index.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
  * General Transitive utilities library
  */
 
+import isNumeric from 'isnumeric/isNumeric'
```
Comment (evansiroky) [inline lib/util/index.js:5]: I'm not a huge fan of adding a whole new package just for this little use here. I'd just go ahead and try to inline some kind of logic in this file.

**Review by evansiroky — CHANGES_REQUESTED** [review]
See comment

File: lib/util/index.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/util/index.js]: I removed the explicit reference in https://github.com/conveyal/transitive.js/pull/62/commits/7e65ad5f2d7327d5930cdae7a71d2c1045f7f8d8. 

Interestingly, `isnumeric` is implicitly referenced by another library (one of CSS libraries it looks like).

**Review by evansiroky — APPROVED** [review]

**Review by landonreed — APPROVED** [review]
Sorry for the delay. This looks good. See recommendation regarding the new story's name. Also, I noticed that changing the segment label size to 0px makes things crash. Not sure if that's from these changes or something else in transitive?

File: stories/Transitive.stories.js [hunk 2]
```diff
@@ -26,6 +26,23 @@ Itinerary.args = {
   styles: undefined
 }
 
+export const Itinerary2 = Template.bind({})
+Itinerary2.args = {
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply stories/Transitive.stories.js:30]: Can you provide steps to reproduce the crash you are observing, please?

[the remark this answers is not in the document]
Comment (landonreed) [reply stories/Transitive.stories.js:30]: 1. Open Itinerary2.
2. Change segment label to `0px`
3. Switch to a different story (needed for re-render after style change)
4. Switch back to Itinerary2.
5. Crash 😢

[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply stories/Transitive.stories.js]: I renamed the story in https://github.com/conveyal/transitive.js/pull/62/commits/448ad95ad9e35a10093aae1f51ed5cf01a4f4c55.

[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply stories/Transitive.stories.js]: Regarding the font size, I tried a few workarounds to set a non-zero label/container size, or avoid rendering text, but that didn't work.

[the remark this answers is not in the document]
Comment (landonreed) [reply stories/Transitive.stories.js]: No prob, can you document your findings in https://github.com/conveyal/transitive.js/issues/66?

**Pull request merged**